core/agents/bug_hunter.py

```diff
@@ -0,0 +1,190 @@
+from enum import Enum
+from typing import Annotated, Literal, Union
+
+from pydantic import BaseModel, Field
+
+from core.agents.base import BaseAgent
+from core.agents.convo import AgentConvo
+from core.agents.response import AgentResponse
+from core.config import magic_words
+from core.db.models.project_state import IterationStatus
+from core.llm.parser import JSONParser
+from core.log import get_logger
+
+log = get_logger(__name__)
+
+
+class StepType(str, Enum):
+    ADD_LOG = "add_log"
+    EXPLAIN_PROBLEM = "explain_problem"
+    GET_ADDITIONAL_FILES = "get_additional_files"
+
+
+class Log(BaseModel):
+    filePath: str
+    referenceCodeSnippet: str = Field(description="Five lines of code before the line where the log needs to be added. Make sure that this contains **ONLY** the code that is currently written in the file. It must not contain the log that you want to add.")
+    log: str
+
+
+class AddLog(BaseModel):
+    type: Literal[StepType.ADD_LOG] = StepType.ADD_LOG
+    logsToAdd: list[Log]
+
+
+class ExplainProblem(BaseModel):
+    type: Literal[StepType.EXPLAIN_PROBLEM] = StepType.EXPLAIN_PROBLEM
+    problem_explanation: str
+
+
+class GetAdditionalFiles(BaseModel):
+    type: Literal[StepType.GET_ADDITIONAL_FILES] = StepType.GET_ADDITIONAL_FILES
+    filePath: str
+
+
+# TODO enable LLM to ask for more files
+class LoggingOptions(BaseModel):
+    decision: Annotated[
+        Union[AddLog, ExplainProblem, GetAdditionalFiles],
+        Field(discriminator="type"),
+    ]
+
+
+class HuntConclusionType(str, Enum):
+    ADD_LOGS = magic_words.ADD_LOGS
+    PROBLEM_IDENTIFIED = magic_words.PROBLEM_IDENTIFIED
+
+
+class HuntConclusionOptions(BaseModel):
+    conclusion: HuntConclusionType = Field(description=f"If more logs are needed to identify the problem, respond with '{magic_words.ADD_LOGS}'. If the problem is identified, respond with '{magic_words.PROBLEM_IDENTIFIED}'.")
+
+
+class BugHunter(BaseAgent):
+    agent_type = "bug-hunter"
+    display_name = "Bug Hunter"
+
+    async def run(self) -> AgentResponse:
+        current_iteration = self.current_state.current_iteration
+
+        if "bug_reproduction_description" not in current_iteration:
+            await self.get_bug_reproduction_instructions()
+        if current_iteration["status"] == IterationStatus.HUNTING_FOR_BUG:
+            # TODO determine how to find a bug (eg. check in db, ask user a question, etc.)
+            return await self.check_logs()
+        elif current_iteration["status"] == IterationStatus.AWAITING_USER_TEST:
+            return await self.ask_user_to_test()
+        elif current_iteration["status"] == IterationStatus.AWAITING_BUG_REPRODUCTION:
+            return await self.ask_user_to_test()
```

Comment:
can we please not do elif and elif and in both of them call same function without any difference? We can at least send some variable to function so that inside function we dont have to check iteration status

Reply:
I added the variable but splitting was so that it's more readable. Let me know if you would prefer going with "or" instead